sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/kafka/KafkaTableProviderTest.java

```diff
@@ -65,17 +64,17 @@ private static Table mockTable(String name) {
     return Table.builder()
         .name(name)
         .comment(name + " table")
-        .location(URI.create("kafka://localhost:2181/brokers?topic=test"))
+        .location("kafka://localhost:2181/brokers?topic=test")
```

Comment:
Why location is not a URI?

Reply:
Location isn't a URL for all table providers, so it is a string. For example, the big query syntax is `[PROJECT_ID]:[DATASET].[TABLE]`. This change comes from the PR this is based on: #4991

Reply:
Got it. The #4991 doesn't show any BQ-specific examples so this wasn't clear. How do we know that the table is a BQ table then if we can't use Location to identify it?

And I think that URI format actually supports the syntax for BQ table IDs, but not sure how semantically correct that would be. It would probably technically be a URN at that point which our code has to resolve using some other data.

Reply:
The DDL already requires a type. Syntax for this looks something like `CREATE TABLE mytable TYPE kafka LOCATION "kafka://localhost:2181/brokers?topic=test"`

PR #4991 links to #4947 which adds bigquery. I was honestly expecting both to be merged before this one was reviewed, but I haven't been able to get reviews on #4991.

Reply:
So we remove the U from URL and then put it back via `TYPE`. Can't object.